### PR TITLE
Update Developer Guide Bookmark

### PR DESCRIPTION
**Pull request by n3wsoldier**
Implementation of Bookmark feature

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -515,6 +516,148 @@ Losing work data can also be a frustrating and costly mistake to translators esp
...
 <br/>
 
+### 4.5 Bookmark Feature
+Overview of the different features provided by bookmark
+The `bookmark` feature aims to provide the user with the ability to create shortcuts to anime they wish to track. This feature further provides tools to monitor the progress of a series and make informative notes.
+
+
+#### 4.5.1 Current Implementation
+
+The Bookmark class uses three ArrayList to store bookmark entries of the user, these arraylists maintain information about the anime index, episode and notes. The synchronisation between arraylist is required so that it enables easy retrieval of bookmark information using the bookmark index on the three arraylist.
```
Comment (ChanJianHao) [inline docs/DeveloperGuide.md:527]: Great work explaining!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -515,6 +516,148 @@ Losing work data can also be a frustrating and costly mistake to translators esp
...
+Given below is the example usage scenario and how the `Bookmark` command behaves at each step.
+
+> :bulb: The command is using one-based indexing while the program stores and recognises the zero-based indexing. 
+The following example will use one-based to be consistent with the command. 
+
+**Step 1:** User inputs command `bookmark`,  the application then calls `Parser#getCommand()` and passes the command to it.
+
+**Step 2:** `BookmarkParser` extracts “Bookmark Action” from the command, and based on the command type, it will determine the respective required field. The `BookmarkParser` is responsible for validating the input parameters and will throw an `AniException` when encountering unrecognised inputs.
+
+For example: `bookmark 2 -e 5`, the parser will create `BookmarkCommand` and call the method setBookmarkAction(“e”), setBookmarkIndex(2) and setBookmarkEpisode(5).
```
Comment (ChanJianHao) [inline docs/DeveloperGuide.md:554]: I like your example

**Review by ChanJianHao — APPROVED** [review]
lgtm!

**Review by EyoWeiChin — APPROVED** [review]
LGTM!

**Pull request merged**